crates/superposition_types/src/custom_query.rs

```diff
@@ -86,3 +91,108 @@ where
         ready(Self::extract_query(req.query_string()).map_err(actix_web::Error::from))
     }
 }
+
+/// Provides struct to extract those query params from the request which are `not wrapped` in contrusts like `platform[param_name]`
+#[derive(Debug, Clone)]
+pub struct DynamicQuery<T: DeserializeOwned>(pub T);
+
+impl<T> CustomQuery for DynamicQuery<T>
+where
+    T: DeserializeOwned,
+{
+    type Inner = T;
+
+    fn regex_pattern() -> &'static str {
+        r"^(?<query_name>[^\[\]]+)$"
+    }
+
+    fn capture_group() -> &'static str {
+        "query_name"
+    }
+
+    fn into_inner(self) -> T {
+        self.0
+    }
+
+    fn new(inner: Self::Inner) -> Self {
+        Self(inner)
+    }
+}
+
```

Comment:
Do we want to filter out the platform query? It doesn't impact resolution in CAC right? json logic will still match more generic contexts